Implementation of gas limits for relaying messages to Linea

Ugo-X opened the pull request:
This PR introduces three new functions to manage gas limits for critical cross-chain operations on the Linea blockchain. These functions allow the contract owner to dynamically adjust gas limits for specific methods, enhancing the flexibility and efficiency of our cross-chain communication system.

Changes:
1. Added `setGasLimitPropagateRoot` function to set the gas limit for the `propagateRoot` method.
2. Added `setGasLimitSetRootHistoryExpiry` function to set the gas limit for the `setRootHistoryExpiry` method.
3. Added `setGasLimitTransferOwnership` function to set the gas limit for the `transferOwnership` method.

closes #3

kfastov commented:
@iamUgo_X Hi! Sorry for making you wait for so long. This issue was mostly theoretical and required some research to determine some initial values for gas limits. That's cool though that you also added the code for setting those limits. I now realize that description was not so clear, and that assigning this task before contracts are ready was a bit dishonest, and I've started to review it too late, so I won't ask you to change anything.
But can you please run `forge fmt` on the code to satisfy the linter? 😊

Ugo-X commented:
> @iamUgo_X Hi! Sorry for making you wait for so long. This issue was mostly theoretical and required some research to determine some initial values for gas limits. That's cool though that you also added the code for setting those limits. I now realize that description was not so clear, and that assigning this task before contracts are ready was a bit dishonest, and I've started to review it too late, so I won't ask you to change anything. But can you please run `forge fmt` on the code to satisfy the linter? 😊

it's no issue, thanks chief, I have run forge fmt and updated.

kfastov commented:
I'll merge this PR after we merge the contracts because they have some common code

Ugo-X commented:
Alright chief, will get to that.

Ugo-X commented:
> @Ugo-X Since contracts are merged now, can you please try to determine the reasonable default gas limits for those methods? You may deploy them on local devnet or testnet or use some online tools like Remix. Don't hesitate to contact me if you need any help

Am I to calculate the exact gas fees?

Ugo-X commented:
Alright

kfastov commented:
@Ugo-X Okay I'll try to get some testing ETH on Sepolia and share with you 🙏

Ugo-X commented:
Alright,thanks 🙏

kfastov commented on src/LineaStateBridge.sol:
```diff
@@ -37,7 +37,7 @@ contract LineaStateBridge is Ownable2Step {
 
     /// @notice The default gas limit amount to buy on an Linea stack chain to do simple transactions
-    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
+    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 988_315;
```
Can you please share how did you get this value? You can provide some screenshots, logs, or just describe it in a few words

Ugo-X replied:
> Can you please share how did you get this value? You can provide some screenshots, logs, or just describe it in a few words

I deployed the contract and used the contract deployed gas limit as the default gas limit.

<img width="1280" alt="Screenshot 2024-07-05 at 22 06 00" src="https://github.com/kfastov/worldcoin-bridge-linea/assets/133219527/07cb78cb-ee83-4aa5-aa8d-e646318f791a">
<img width="1280" alt="Screenshot 2024-07-05 at 22 06 10" src="https://github.com/kfastov/worldcoin-bridge-linea/assets/133219527/45d1b1ae-f153-44bf-9d13-081216363acf">

LINK:  https://sepolia.lineascan.build/tx/0xd6ddc75572298d7d6c69ab9bc21227ef45067856016bcde294f6f4f166ab19e5

kfastov replied:
@Ugo-X Thanks for deploying the contracts on testnet and sharing all the details!
However it seems that you estimated gas limit for the deploy transaction, not the method calls.
Also, I think `1_000_000` is a good first approximation for `988_315`. We'll figure out the actual values when we will test the contracts on testnet

kfastov commented on bun.lockb:
Hmm that lock file changes constantly, and even its attributes do. I should read the bun's best practices 🤔

kfastov commented on src/LineaStateBridge.sol:
```diff
@@ -37,7 +37,7 @@ contract LineaStateBridge is Ownable2Step {
 
     /// @notice The default gas limit amount to buy on an Linea stack chain to do simple transactions
-    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
+    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 988_315;
```
I propose to just leave it as is and add a comment reminding of this work
```suggestion
    // estimated by @Ugo-X as ~988_315
    uint32 public constant DEFAULT_LINEA_GAS_LIMIT = 1_000_000;
```

kfastov reviewed (changes requested):
Approve with comments above

Ugo-X commented:
reverted back to the original Approximate gas limit. How should i proceed?
